src/app/shared/post-it-collections/post-it-collection-model.ts

```diff
@@ -17,11 +17,11 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import {Collection} from '../../core/dto';
+import {CollectionModel} from '../../core/store/collections/collection.model';
 
 export class PostItCollectionModel {
 
-  public collection: Collection;
+  public collection: CollectionModel;
   public initializing: boolean;
   public initialized: boolean;
```

Comment:
We still need the `initializing` flag to make sure we won't dispatch `Update` or `Create` during the creation multiple times.

Reply:
You can make use of the existing properties in `CollectionModel`. Basically, what you want to do is to distinguish between three collection states:
1. just created in component, not sent to server
2. sent to server, no response yet
3. received instance from the server

The collection would in those states if the following conditions are met:
1. neither `id` nor `correlationId` set
2. `correlationId` set but `id` not set
3. `id` set

Would it be possible to implement it like this? But it is probably a task for another PR.

Reply:
Sure. It would be possible.

Just a question though.. the `correlationId` is `Id` for the frontend side? If I set it to some value.. the store won't change it to different value?

Reply:
`id` is set by the server when the collection is actually created in the database. `correlationId` is just a front-end identifier which is not propagated to the server. It is not created automatically. You have to create it when dispatching an action to create a new collection and it will be kept in the store with that collection (even when the fresh instance from the server is received) unless you clear the store (application refresh, change of project, etc.).